energy-integration/governance-v2/src/views.rs

```diff
@@ -45,6 +45,19 @@ pub trait ViewsModule:
         }
     }
 
+    #[view(getUserVotingPower)]
+    fn get_user_voting_power(&self, user_address: ManagedAddress) -> BigUint {
+        let user_quorum = self.get_energy_amount_non_zero(&user_address);
+        user_quorum.sqrt()
```

Comment:
i would declare a method smoothing_function and use that everywhere. If the smoothing function will change there is only one place where the change will be applied